applications/luci-app-wireguard/htdocs/luci-static/resources/view/wireguard/status.js

```diff
@@ -0,0 +1,223 @@
+	} else {
+		ago = _('over a day ago');
+	}
+	var t = new Date(timestamp * 1000);
+	return t.toUTCString() + ' (' + ago + ')';
+}
+
+function generatePeerOption(key, title, value) {
+	return E('div', { 'class': 'cbi-value', 'style': 'padding: 0;' }, [
+		E('label', {
+			'class': 'cbi-value-title', 'style': 'font-weight: bold;'
+		}, title),
+		E('input', {
+			'class': 'cbi-input-text',
+			'data-name': key,
+			'style': 'border: none; float: left; width: 50%;',
+			'disabled': '',
+			'value': value
+		})
+	]);
+}
+
+function generatePeerTable(options, iconSrc) {
+	return E('div', { 'class': 'table cbi-section-table' }, [
+		E('div', { 'class': 'td' },
+			E('img', { 'src': iconSrc, 'class': 'tunnel-icon' })
+		),
+		E('div', { 'class': 'td peer-options' },
+			options.filter(function (option) {
+				return option[2] != null;
+			}).map(function (option) {
+				return generatePeerOption(...option);
+			})
+		)
+	]);
+}
+
+function getTunnelIcon(latestHandshake) {
+	var img = (new Date().getTime() / 1000 - latestHandshake) < 140 ?
+		'tunnel' : 'tunnel_disabled';
+
+	return L.resource('icons', img + '.png');
+}
+
+function generatePeerRows(peers) {
+	var peerRows = [];
+
+	peers.forEach(function (peer) {
+		var peerData = parsePeerData(peer);
+		var iconSrc = getTunnelIcon(peer.latest_handshake);
+
+		peerRows.push(E('div', {
+			'class': 'tr cbi-section-table-row'
+		}, [
+			E('div', {
+				'class': 'td peer-name',
+				'style': 'width: 25%; font-size: 0.9rem;'
+			}, peer.name),
+			E('div', { 'class': 'td', 'data-section-id': peer.name },
+				generatePeerTable(peerData, iconSrc)
+			)
+		]));
+	});
+
+	return peerRows;
+}
+
+function parseIfaceData(iface) {
+	return [
+		['public_key', _('Public Key'),
+			iface.public_key != '(none)' ? iface.public_key : null],
+		['listen_port', _('Listen Port'),
+			iface.listen_port > 0 ? iface.listen_port : null],
+		['fwmark', _('Firewall Mark'),
+			iface.fwmark != 'off' ? iface.fwmark : null]
+	];
+}
+
+function parsePeerData(peer) {
+	return [
+		['public_key', _('Public Key'),
+			peer.public_key],
+		['endpoint', _('Endpoint'),
+			peer.endpoint == '(none)' ? null : peer.endpoint],
+		['allowed_ips', _('Allowed IPs'),
+			peer.allowed_ips.length == 0 ? null : peer.allowed_ips.join('\n')],
+		['persistent_keepalive', _('Persistent Keepalive'),
+			peer.persistent_keepalive == 'off' ? null : peer.persistent_keepalive + 's'],
+		['latest_handshake', _('Latest Handshake'),
+			timestampToStr(peer.latest_handshake)],
+		['transfer_rx', _('Data Received'),
+			bytesToStr(peer.transfer_rx)],
+		['transfer_tx', _('Data Transmitted'),
+			bytesToStr(peer.transfer_tx)]
+	];
+}
+
+return view.extend({
+	load: function () {
+		return callGetWgInstances();
+	},
+	poll_status: function (nodes, ifaces) {
+		Object.keys(ifaces).forEach(function (ifaceName) {
+			var iface = ifaces[ifaceName];
+
+			var section = nodes.querySelector(
+				'[data-section-id="' + ifaceName + '"]'
```

Comment:
This and similar string concat expressions could be replaced with `nodes.querySelector('[data-section-id="%q"]'.format(ifaceName))` for slightly better readability.